scripts/python/packages/parthenon_tools/parthenon_tools/phdf.py

```diff
@@ -81,106 +81,143 @@ def __init__(self, filename):
+        except:
+            self.IncludesGhost = 0
+        self.NumBlocks = info.attrs["NumMeshBlocks"]
+        self.MeshBlockSize = info.attrs["MeshBlockSize"]
+        try:
+            self.BlocksPerPE = info.attrs["BlocksPerPE"]
+        except:
+            self.BlocksPerPE = np.array((1), self.NumBlocks)
+        self.Coordinates = info.attrs["Coordinates"]
+        self.CellsPerBlock = np.prod(self.MeshBlockSize)
+        self.TotalCells = self.NumBlocks * self.CellsPerBlock
+
+        # Read in Params (older output files don't have this, so make it optional)
+        if "Params" in f:
+            self.Params = dict(f["Params"].attrs)
+        else:
+            self.Params = None
+
+        # Read in coordinates
+        def load_coord(coord_i):
+            coord_name = ["x", "y", "z"][coord_i]
+
+            tmp = f["/Locations/" + coord_name][:, :]
+            vol_loc = "/VolumeLocations/" + coord_name
+            if vol_loc in f:
+                coord = f[vol_loc][:, :]
             else:
-                self.Params = None
-
-            # Read in coordinates
-            def load_coord(coord_i):
-                coord_name = ["x", "y", "z"][coord_i]
-
-                tmp = f["/Locations/" + coord_name][:, :]
-                vol_loc = "/VolumeLocations/" + coord_name
-                if vol_loc in f:
-                    coord = f[vol_loc][:, :]
-                else:
-                    coord = np.zeros((self.NumBlocks, self.MeshBlockSize[coord_i]))
-                    for bId in range(self.NumBlocks):
-                        for cId in range(self.MeshBlockSize[coord_i]):
-                            coord[bId, cId] = 0.5 * (tmp[bId, cId] + tmp[bId, cId + 1])
-                coordf = tmp
-                return tmp, coord, coordf
-
-            tmpx, self.x, self.xf = load_coord(0)
-            tmpy, self.y, self.yf = load_coord(1)
-            tmpz, self.z, self.zf = load_coord(2)
-
-            # fill in self.offset and block bounds
-            self.offset = [0, 0, 0]
-            for i in range(3):
-                if self.MeshBlockSize[i] > 1:
-                    self.offset[i] = self.NGhost * self.IncludesGhost
-
-            # fill in self.BlockBounds
-            self.BlockBounds = [None] * self.NumBlocks
-            xo = self.NGhost * self.IncludesGhost
-            iOffsets = [
-                self.offset[0],
-                self.MeshBlockSize[0] - self.offset[0],
-                self.offset[1],
-                self.MeshBlockSize[1] - self.offset[1],
-                self.offset[2],
-                self.MeshBlockSize[2] - self.offset[2],
+                coord = np.zeros((self.NumBlocks, self.MeshBlockSize[coord_i]))
+                for bId in range(self.NumBlocks):
+                    for cId in range(self.MeshBlockSize[coord_i]):
+                        coord[bId, cId] = 0.5 * (tmp[bId, cId] + tmp[bId, cId + 1])
+            coordf = tmp
+            return tmp, coord, coordf
```

Comment:
Aren't tmp and coordf always going to be the same, why return three items when you can return two, is it because it makes it easier to split up the return values.